services/community-site/src/utils/validators.ts

```diff
@@ -0,0 +1,34 @@
+import { Validator, ValidatorWithStats, YieldedValidator } from '../interfaces/Validator';
+
+const calulcateValidatorYield = (validators: Validator[]): ValidatorWithStats[] | Validator[] => {
+  const hasStats = (validators[0] as ValidatorWithStats)?.pbftsProduced !== 0;
+  if (!hasStats) {
+    return validators;
+  }
+  const validatorsWithStats = validators as ValidatorWithStats[];
```

Comment:
You're upcasting here. It's better to downcast.

Reply:
Yeah, that would be feasible, but then I'll need to change some of the business logic of the whole flow, starting with `useExplorerStats`, I'll do that for the next iteration